compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -861,7 +861,7 @@ object Contexts {
                        with Phases.PhasesBase
                        with Plugins {
 
-    val settings: ScalaSettings = ScalaSettings
+    val settings: ScalaSettings.type = ScalaSettings
```

Comment:
Now that this is a singleton we could use `ScalaSettings` instead of `settings`. Not sure what this val gives us. Same for `def settings: ScalaSettings.type = base.settings`.

I know that would imply a massive refactoring of all `ctx.settings.XYZ.value` to `ScalaSettings.XYZ.value`.

Reply:
> I know that would imply a massive refactoring of all ctx.settings.XYZ.value to ScalaSettings.XYZ.value.

That's the reason. There is no cost to keeping it a setting right now, and the PR is getting quite big. This is to be done in the next PR, so we don't block the ScalaCLI integration with it.